### ISSUE TITLE
hasattr(obj, beam) causes exception

### ISSUE THREAD

**Issue by keflavich** [radio-astro-tools/spectral-cube#719]
Regression introduced by https://github.com/radio-astro-tools/spectral-cube/pull/700, I believe:

```python
ipdb> hasattr(obj,'beam')
*** spectral_cube.utils.NoBeamError: No beam is defined for this SpectralCube or the beam information could not be parsed from the header. A `~radio_beam.Beam` object can be added using `cube.with_beam`.
```

This is because you can't access a .beam attribute without raising an exception if it's not defined.  But, the object will always have the `.beam` attribute.

**Comment by keflavich** [comment]
I have a fix in progress

**Comment by e-koch** [comment]
Fixed in #720
